### ISSUE TITLE
[Service Bus Client] Refactor Diagnostic Tests

### ISSUE THREAD

**Issue by jsquire** [Azure/azure-sdk-for-net#5692]
# Summary

The family of diagnostic tests for the Service Bus client library make some very strong assumptions about the order of events, content of them, and exclusivity of those events in the shared environment.  The initial version of these tests, before their adoption into the central SDK repository, required that tests run serially in order to protect these assumptions.  

As part of the workstream to migrate into the central repository, the test infrastructure was refactored in order to better isolate tests from one another to prevent cascading failures, improve overall reliability, and allow for tests to run in parallel.

In support of these efforts, the test infrastructure around diagnostics were enhanced to allow filtering of diagnostic information to a specific Service Bus entity (queue, topic, subscription) that was of interest to an individual test.  This allowed the tests to continue to behave as if they were running in isolation despite parallel execution.

The drawback to the approach used is that any unexpected events emitted to the environment which are not bound to the specific entities that the tests have expressed an interest in are not seen, which may hide an issue with the client generating malformed or extraneous diagnostics information.

# Scope of Work

Implement an approach to allow for tests in the diagnostic family to detect malformed or extraneous diagnostics information while continuing to provide consistent results without the need to run tests serially.   It is permissible to force serialization of the diagnostics family of tests, though it is preferred to have them continue to run in parallel, if possible.

# Out of Scope

- Refactoring of the general approach for testing, the test cases, code style, and other artifacts of the codebase; changes should be kept as tightly scoped and as minimal as possible.  

- Adding of additional test cases or new testing scenarios.  

# Success Criteria

- The diagnostic tests are able to detect malformed and extraneous diagnostics information generated by the client;  this may happen within the scope of existing tests or with the addition of new tests dedicated to that goal.  

- The diagnostic tests continue to provide reliable and stable test results for nightly runs; there are no frequent and inconsistent failures.  

- The results of the diagnostic tests are not influenced by outside factors nor do they impact other tests.  Any failures do not cascade.

- Tests other than the diagnostic tests may be run in parallel without impacting the diagnostic tests.  While it is permissible to force serial execution on the diagnostics tests, it is preferred that they continue to support running in parallel.  

# Details

### Diagnostic Test Classes

- Diagnostics\ExtractActivityTests.cs
- Diagnostics\QueueClientDiagnosticTests.cs
- Diagnostics\SessionDiagnosticTests.cs
- Diagnostics\SubscriptionDiagnosticTests.cs
- Infrastructure\FakeDiagnosticsListenerTests.cs

### Diagnostic Test Infrastructure

- **Diagnostics\DiagnosticTests.cs **
   _This is the base class for the diagnostic live test family, responsible for providing functionality to setup the event listener and related infrastructure.  `CreateEventListener` is an area of interest._  

- **Diagnostics\FakeDiagnosticsListener.cs**  
  _Provides a minimal listener for diagnostic events, including the first pass of entity filtering.  The constructor sets up the initial entity predicate and may be of interest._  

### Considerations

The initial thought is to consider making use of the XUnit [`CollectionDefinition`](https://xunit.net/docs/shared-context.html) attribute to group the diagnostics tests into a single collection and to set `DisableParallelization = true`.   In theory, this should create a serialized set of execution for the diagnostic tests which are run in isolation after the rest of the tests are run in parallel.
